include/flash.h

```diff
@@ -127,6 +165,47 @@ static inline int flash_write_protection_set(struct device *dev, bool enable)
 	return api->write_protection(dev, enable);
 }
 
+#if defined(CONFIG_FLASH_PAGE_LAYOUT)
+
+struct flash_pages_info {
+	off_t start_offset;
+	size_t size;
+};
+
+/**
+ *  @brief  Get size and start offset of flash page at certain flash offset.
```

Comment:
It is offset form the base of flash. It is possible to turn it to absolute. Then flash api needs to be extended more for retrieve the base address (that because it is possible to use at last 2 driver concurrently one for embedded flash and one for a external flash like W25QXXDV_SPI).